### ISSUE TITLE
False positive: arguments-differ for __init_subclass__ in generic

### ISSUE THREAD

**Issue by ShadowLNC** [pylint-dev/pylint#8919]
### Bug description

```python
from typing import Generic, TypeVar


T = TypeVar("T")


class Testing(Generic[T]):  # pylint: disable=too-few-public-methods
    """Placeholder."""

    def __init_subclass__(cls, /, **kwargs: Any) -> None:
        """Placeholder."""
        super().__init_subclass__(**kwargs)
```

### Command used

```shell
pylint file.py
```

### Pylint output

```shell
************* Module file
file.py:10:4: W0221: Variadics removed in overriding 'Testing.__init_subclass__' method (arguments-differ)

------------------------------------------------------------------
Your code has been rated at 9.88/10 (previous run: 9.88/10, +0.00)
```

### Expected behavior

I believe this is a false positive.

When I don't inherit from `Generic[T]`, the error goes away.

### Pylint version

```shell
pylint 2.17.5
astroid 2.15.6
Python 3.11.1 (v3.11.1:a7a450f84a, Dec  6 2022, 15:24:06) [Clang 13.0.0 (clang-1300.0.29.30)]
```

### OS / Environment

MacOS Ventura 13.5

**Comment by mbyrnepr2** [comment]
Thanks!
